tests/baselines/reference/constEnumNamespaceReferenceCausesNoImport2.js

```diff
@@ -0,0 +1,50 @@
+//// [tests/cases/compiler/constEnumNamespaceReferenceCausesNoImport2.ts] ////
+
+//// [foo.ts]
+export module ConstEnumOnlyModule {
+  export const enum ConstFooEnum {
+    Some,
+    Values,
+    Here
+  }
+}
+
+//// [reexport.ts]
+import * as Foo from "./foo";
+export = Foo.ConstEnumOnlyModule;
+
+//// [index.ts]
+import Foo = require("./reexport");
+function check(x: Foo.ConstFooEnum): void {
+  switch (x) {
+    case Foo.ConstFooEnum.Some:
+      break;
+  }
+}
+
+//// [foo.js]
+"use strict";
+exports.__esModule = true;
+exports.ConstEnumOnlyModule = void 0;
+var ConstEnumOnlyModule;
+(function (ConstEnumOnlyModule) {
+    var ConstFooEnum;
+    (function (ConstFooEnum) {
+        ConstFooEnum[ConstFooEnum["Some"] = 0] = "Some";
+        ConstFooEnum[ConstFooEnum["Values"] = 1] = "Values";
+        ConstFooEnum[ConstFooEnum["Here"] = 2] = "Here";
+    })(ConstFooEnum = ConstEnumOnlyModule.ConstFooEnum || (ConstEnumOnlyModule.ConstFooEnum = {}));
+})(ConstEnumOnlyModule = exports.ConstEnumOnlyModule || (exports.ConstEnumOnlyModule = {}));
+//// [reexport.js]
+"use strict";
+var Foo = require("./foo");

```

Comment:
A bit of a separate bug, but closely inspecting `checkPropertyAccessExpressionOrQualifiedName` led me to discover that this import was missing until I added the check for `isExportOrExportExpression`.